lib/remote_participant_view.dart

```diff
@@ -0,0 +1,236 @@
+// © 2022 Daily, Co. All Rights Reserved
+
+import 'dart:async';
+
+import 'package:daily_flutter/daily_flutter.dart';
+import 'package:flutter/material.dart';
+
+class RemoteParticipantView extends StatefulWidget {
+  const RemoteParticipantView({
+    super.key,
+    required this.client,
+    required this.participantId,
+    this.size = const Size.square(160),
+  });
+
+  final CallClient client;
+  final ParticipantId participantId;
+  final Size size;
+
+  @override
+  State<RemoteParticipantView> createState() => _RemoteParticipantViewState();
+}
+
+class _RemoteParticipantViewState extends State<RemoteParticipantView> {
+  String? _username;
+  bool _isCameraMuted = false;
+  bool _isMicMuted = false;
+  StreamSubscription? _eventSubscription;
+  final _controller = VideoViewController();
+
+  @override
+  void didUpdateWidget(RemoteParticipantView oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _controller.setTrack(widget.client.participants.all[widget.participantId]?.media?.camera.track);
+    _maybeUpdate(widget.client.participants.remote[widget.participantId]);
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _maybeUpdate(widget.client.participants.remote[widget.participantId]);
+    _eventSubscription = widget.client.events.listen((event) {
+      if (!mounted) return;
+      event.whenOrNull<void>(
+        subscriptionsUpdated: (_) {
+          _maybeUpdate(widget.client.participants.remote[widget.participantId]);
+        },
+        participantJoined: (participant) async {
+          if (participant.id == widget.participantId) {
+            _maybeUpdate(participant);
+          }
+        },
+        participantUpdated: (participant) async {
+          if (participant.id == widget.participantId) {
+            _maybeUpdate(participant);
+          }
+        },
+      );
+    });
+  }
+
+  void _maybeUpdate(Participant? participant) => setState(() {
+        _username = participant?.info.username;
+        _isMicMuted = participant?.isMicrophoneMuted ?? false;
+        _isCameraMuted = participant?.isCameraMuted ?? false;
+        _controller.setTrack(participant?.media?.camera.track);
+      });
+
+  @override
+  void dispose() {
+    _eventSubscription?.cancel();
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    late final permissions = widget.client.participants.all[widget.participantId]?.info.permissions;
+    late final canSend = permissions?.canSend;
+
+    late final canSendMic = canSend?.contains(CanSendPermission.microphone) ?? true;
```

Comment:
Question for my own curiosity! When would `canSend` be legitimately null or otherwise falsy (whatever the Dart equivalent is)? Is it safe to default to `true` in that scenario?